agent.go

```diff
@@ -952,6 +952,35 @@ func (c *container) closeProcessPipes(pid string) {
 	}
 }
 
+func (c *container) closeConsoleSocket(pid string) error {
+	cid := c.container.ID()
+	proc := c.getProcess(pid)
+
+	errMsg := "Could not close process socket pair"
+
+	if proc == nil {
+		return fmt.Errorf("Container %s process %s no longer exists", cid, pid)
+	}
+
+	if proc.consoleSock != nil {
+		if err := proc.consoleSock.Close(); err != nil {
+			return fmt.Errorf(errMsg)
```

Comment:
You don't need `Errorf()` as written. But how about making it as follows (here and below)?

```
return fmt.Errorf("%s: %s", errMsg, err)
```

Reply:
(By that I mean you could just use `errors.New(errMsg)`).

Reply:
Yes let me change that ;)